ISISComputingGroup/ibex_gui
# Connecting to p vs

###Description of work

Work done in this branch with previous tickets (already reviewed):
* Hooked up the GUI to the `GET_SCREENS` PV from the blockserver in the `devicescreens` plugin
* Created a `ui.devicescreens` plugin with a mock UI, just for test/review purpose until the real GUI for this plugin is setup

As part of ticket #1122:
* Hooked up the GUI to the `SET_SCREENS` PV from the blockserver.
* Refactored existing classes of `devicescreens` to follow more closely what is done for the synoptic plugin (e.g. `BundleActivator`)
* GUI makes use of the schema provided by the blockserver
* Unit tests for all relevant classes in `devicescreens` plugin

###To test

Ticket #1122

NOTE: This branch introduces new plugins. When starting the GUI, a "Device Screens" button should appear on the list of perspective on the left of the GUI. If this doesn't happen, from within Eclipse's Package Explorer clear your workspace and re-import the entire project (and set target platform).

In Devices Screens in the GUI, you will see a label showing the contents of your `GET_SCREENS` PV (already there from already reviewed #1121). There is also a text box for you to enter a new xml for the `SET_SCREENS` PV. An example content is shown below.

###Acceptance criteria

When entering a valid XML in the text box, the same XML appears in the label.
This demonstrates that the entire loop is closed correctly:
* the text from the text box is parsed into a `DeviceScreensDescription` object
* that object is re-converted to XML and written to the `SET_SCREENS` PV (following the schema)
* the blockserver takes the `SET_SCREENS` PV content and copies it into the `GET_SCREENS` PV
* the GUI reads the `GET_SCREENS` PV, and parses it into a new `DeviceScreensDescription` object
* that object is converted to an xml string for display purposes and shown in the label in the GUI

Example XML:
```
<?xml version="1.0" ?>
<devices xmlns="http://epics.isis.rl.ac.uk/schema/screens/1.0/">
    <device>
        <name>MyDevice</name>
        <key>MyDeviceKey</key>
        <type>OPI</type>
        <properties>
            <property>
                <key>Property1</key>
                <value>Value1</value>
            </property>
            <property>
                <key>Property2</key>
                <value>Value2</value>
            </property>
        </properties>
    </device>
</devices>
```

---

#### Code Review


### Functional Tests
